greenmail-core/src/main/java/com/icegreen/greenmail/store/SimpleMessageAttributes.java

```diff
@@ -381,7 +380,7 @@ String parseEnvelope() {
         }
         response.add(SP);
         if (messageID != null && messageID.length > 0) {
-            messageID[0] = JavaEscape.escapeJava(messageID[0]);
+            messageID[0] = escapeHeader(messageID[0]);
             response.add(Q + messageID[0] + Q);

```

Comment:
See comment for escapeHeader: Obsolete surrounding with quotation Q  if additionally using MimeUtility.quote